EU4toV2/Source/EU4World/World.cpp

```diff
@@ -142,38 +117,26 @@ EU4::world::world(const string& EU4SaveFileName, const mappers::IdeaEffectMapper
 		std::regex("countries"),
 		[this, ideaEffectMapper](const std::string& countriesText, std::istream& theStream)
 		{
+			LOG(LogLevel::Info) << "- Loading Countries";
 			loadCountries(theStream, ideaEffectMapper);
 		}
 	);
-	registerKeyword(std::regex("diplomacy"), [this](const std::string& diplomacyText, std::istream& theStream) {
-		auto diplomacyObject = commonItems::convert8859Object(diplomacyText, theStream);
-		loadDiplomacy(diplomacyObject);
+	registerKeyword(std::regex("diplomacy"), [this](const std::string& unused, std::istream& theStream) {
+		LOG(LogLevel::Info) << "- Loading Diplomacy";
+		EU4::EU4Diplomacy theDiplomacy(theStream);
+		diplomacy = theDiplomacy.getAgreements();
+		LOG(LogLevel::Info) << "- Loaded " << diplomacy.size() << " agreements";
 	});
-	registerKeyword(std::regex("map_area_data"), [this](const std::string& mapAreaKey, std::istream& theStream) {
-		// Add the missing equals sign to help out the object parser.
-		auto mapAreaObject = commonItems::convert8859Object(mapAreaKey + "=", theStream);
-		auto vec = mapAreaObject->getValue(mapAreaKey);
-		if (0 < vec.size())
-		{
-			loadMapAreaData(vec[0]);
-		}
-		else
-		{
-			LOG(LogLevel::Warning) << "Could not find map area "
-			                          "data, state information "
-			                          "will not be loaded.";
-		}
+	registerKeyword(std::regex("map_area_data"), [this](const std::string& unused, std::istream& theStream) {
```

Comment:
As a historical note, this is the section that motivated the new parser. Paradox kept breaking the map_area_data in new and interesting ways - at one point the best strategy was to edit the save before parsing it!

The potential to remove Boost and the resulting configuration nightmares was a bonus. As were the speed increases I saw. And the fact that a debug build becomes practical.

Reply:
Indeed. However for our purposes, this section is worthless as prosperity does not map to anything tangible.